src/cargo/core/compiler/build_config.rs

```diff
@@ -167,37 +163,28 @@ pub enum MessageFormat {
     Short,
 }
 
-/// The general "mode" for what to do.
-///
-/// This is used for two purposes. The commands themselves pass this in to
-/// `compile_ws` to tell it the general execution strategy. This influences
-/// the default targets selected. The other use is in the `Unit` struct
-/// to indicate what is being done with a specific target.
+/// The specific action to be performed on each `Unit` of work.
 #[derive(Clone, Copy, PartialEq, Debug, Eq, Hash, PartialOrd, Ord)]
 pub enum CompileMode {
-    /// A target being built for a test.
+    /// Test with `rustc`.
     Test,
-    /// Building a target with `rustc` (lib or bin).
+    /// Compile with `rustc`.
     Build,
-    /// Building a target with `rustc` to emit `rmeta` metadata only. If
-    /// `test` is true, then it is also compiled with `--test` to check it like
+    /// Type-check with `rustc` by emitting `rmeta` metadata only.
+    ///
+    /// If `test` is true, then it is also compiled with `--test` to check it like
     /// a test.
     Check { test: bool },
-    /// Used to indicate benchmarks should be built. This is not used in
-    /// `Unit`, because it is essentially the same as `Test` (indicating
-    /// `--test` should be passed to rustc) and by using `Test` instead it
-    /// allows some de-duping of Units to occur.
-    Bench,
-    /// A target that will be documented with `rustdoc`.
-
+    /// Document with `rustdoc`.
+    ///
     /// If `deps` is true, then it will also document all dependencies.
     /// if `json` is true, the documentation output is in json format.
     Doc { deps: bool, json: bool },
```

Comment:
Why is the deps field on CompileMode too and not just on UserIntent?

Reply:
I missed that. You're right.

Reply:
Went ahead and removed the `deps` field as well. Thanks for the good catch!